sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -145,18 +145,39 @@ case class UnaryPositive(child: Expression)
   """,
   since = "1.2.0",
   group = "math_funcs")
-case class Abs(child: Expression)
+case class Abs(child: Expression, failOnError: Boolean = SQLConf.get.ansiEnabled)
```

Comment:
Does the interpreted code has the same behaviour?

Reply:
Yes, it happens on `private lazy val numeric = TypeUtils.getNumeric(dataType, failOnError)`
It will use exact version of `negate` eventually. 
Otherwise, it won't pass unit tests.